src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -355,30 +355,54 @@
        vec
        (sorted-map :name environment :modules)))
 
+(def ModuleList
+  {schema/Str
+   schema/Str})
+
+(schema/defn
+  extract-module-data :- [{schema/Str schema/Str}]
+  [modules]
+  (->> modules
+       (map #(get-in % [:modules]))
+       (apply concat)
+       distinct
+       (map vals)))
+
 (schema/defn ^:always-validate
   environment-module-response! :- ringutils/RingResponse
   "Process the environment module information, returning a ring response to be
   propagated back up to the caller of the environment_modules endpoint."
-  [info-from-jruby :- List
-   environment :- schema/Str]
-  (let [info-as-json (module-info-from-jruby->module-info-for-json
-                       info-from-jruby environment)]
-    (middleware-utils/json-response 200 info-as-json)))
+  ([info-from-jruby :- Map]
+   (let [all-info-as-json (map #(module-info-from-jruby->module-info-for-json
+                                  (val %)
+                                  (name (key %)))
+                               (sort-nested-info-maps info-from-jruby))]
+     (println (extract-module-data all-info-as-json))
```

Comment:
Since this just appears to be for debugging, I think we should remove this line - and the corresponding `extract-module-data` function since nothing else appears to use it - before merging.